lib/rules/no-implied-eval.js

```diff
@@ -28,141 +35,112 @@ module.exports = {
          */
-        function isImpliedEvalCallExpression(node) {
-            const isMemberExpression = (node.callee.type === "MemberExpression"),
-                isIdentifier = (node.callee.type === "Identifier"),
-                isImpliedEvalCallee =
-                    (isIdentifier && CALLEE_RE.test(node.callee.name)) ||
-                    (isMemberExpression && isImpliedEvalMemberExpression(node.callee));
-
-            return isImpliedEvalCallee && node.arguments.length;
+        function isSpecifiedIdentifier(node, specifiers) {
+            return node.type === "Identifier" && specifiers.includes(node.name);
         }
 
         /**
-         * Checks that the parent is a direct descendent of an potential implied eval CallExpression, and if the parent is a CallExpression, that we're the first argument.
-         * @param {ASTNode} node The node to inspect the parent of.
-         * @returns {boolean} Was the parent a direct descendent, and is the child therefore potentially part of a dangerous argument?
-         * @private
+         * Checks a given node is a MemberExpression node which has the specified name's
+         * property.
+         * @param {ASTNode} node A node to check.
+         * @param {string[]} specifiers Array of specified name.
+         * @returns {boolean} `true` if the node is a MemberExpression node which has
+         *      the specified name's property
          */
-        function hasImpliedEvalParent(node) {
-
-            // make sure our parent is marked
-            return node.parent === last(last(impliedEvalAncestorsStack)) &&
-
-                // if our parent is a CallExpression, make sure we're the first argument
-                (node.parent.type !== "CallExpression" || node === node.parent.arguments[0]);
+        function isSpecifiedMember(node, specifiers) {
+            return node.type === "MemberExpression" && specifiers.includes(astUtils.getStaticPropertyName(node));
         }
 
         /**
-         * Checks if our parent is marked as part of an implied eval argument. If
-         * so, collapses the top of impliedEvalAncestorsStack and reports on the
-         * original CallExpression.
-         * @param {ASTNode} node The CallExpression to check.
-         * @returns {boolean} True if the node matches, false if not.
-         * @private
+         * Reports if the `CallExpression` node has evaluated argument.
+         * @param {ASTNode} node A CallExpression to check.
+         * @returns {void}
          */
-        function checkString(node) {
-            if (hasImpliedEvalParent(node)) {
+        function reportImpliedEvalCallExpression(node) {
+            const [firstArgument] = node.arguments;
 
-                // remove the entire substack, to avoid duplicate reports
-                const substack = impliedEvalAncestorsStack.pop();
+            const staticValue = getStaticValue(firstArgument, context.getScope());
```

Comment:
I think that `getStaticValue` doesn't expect `undefined` as the node ([get-static-value.js#L507](https://github.com/mysticatea/eslint-utils/blob/master/src/get-static-value.js#L507)). This works at the moment, but it might be more correct to check first if the `CallExpression` has arguments?

Reply:
@mdjermanovic oh yes. That makes sense. :) Thanks